src/client_side_request.cc

```diff
@@ -75,6 +75,7 @@
 #endif
 
 #if FOLLOW_X_FORWARDED_FOR
+#define MAX_X_FORWARDED_HOP 16
```

Comment:
If this is intended for build-time tuning this `#define` needs to be wrapped to allow builders value to overwrite this default.

```suggestion
#if !defined(MAX_X_FORWARDED_HOP)
#define MAX_X_FORWARDED_HOP 16
#endif
```

Otherwise,
  If not the limit should be set as ` static const size_t` at the point in the code just before it is used. see other comment for that.

Reply:
I prefer allowing builders to change the `MAX_X_FORWARDED_HOP` value. I am not sure if the default value is high enough not to break some usages. What do you think?

Reply:
I changed the default to 64. Please see branch commit 596ae3d for rationale.